src/core/surface/completion_queue.h

```diff
@@ -39,6 +39,17 @@
 #include "src/core/iomgr/pollset.h"
 #include <grpc/grpc.h>
 
+typedef struct grpc_cq_completion {
+  /** user supplied tag */
+  void *tag;
+  /** done callback - called when this queue element is no longer
+      needed by the completion queue */
+  void (*done)(void *done_arg, struct grpc_cq_completion *c);
+  void *done_arg;
+  /** next pointer; low bit is used to indicate success or not */
```

Comment:
It's possibly a little premature, but generally:
- we need to keep these things pre-allocated on grpc_call
- grpc_call is way too big
- let's get the obvious stuff done as we're doing work

I expect some future work to get ride of the done() member here at some point also.